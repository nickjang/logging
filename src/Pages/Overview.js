import React, {Component} from 'react';
// just do content
class Overview extends Component {
  render() {
    return null;
  }
}

export default Overview;