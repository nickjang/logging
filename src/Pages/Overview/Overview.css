.overview h2 {
  text-align: center;
}

.overview {
  display: flex;
  flex-direction: column;
  width: fit-content;
  margin: 2rem auto;
  padding: 0 1rem;
  align-items: center;
}

.overview p {
  font-family: var(--note-font);
}
