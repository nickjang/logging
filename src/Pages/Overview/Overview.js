import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import LoggingContext from '../../Context/LoggingContext';
import './Overview.css';

class Overview extends Component {
  static contextType = LoggingContext;

  handleNext = (e) => {
    e.preventDefault();
    this.props.history.push('/getting-started');
  }

  render() {
    if (this.context.account.email) this.props.history.push('/');
    
    return (
      <main>
        <article>
          <h2>Welcome to the logging app!</h2>
          <p>You can create logs by starting and stopping the clock.</p>
          <p>View your logs by clicking on the View page at the top.</p>
          <p>Highlight logs you want to adjust or export, also from the View page.</p>
          <button onClick={(e) => { this.handleNext(e) }}>Next</button>
        </article>
      </main>
    );
  }
}

export default withRouter(Overview);