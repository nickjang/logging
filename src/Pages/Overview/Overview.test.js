import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Overview from './Overview';

describe(`Overview component`, () => {
  it('renders Overview by default', () => {
    const wrapper = shallow(<Overview />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})