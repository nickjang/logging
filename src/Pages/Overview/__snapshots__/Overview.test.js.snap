// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview component renders Overview by default 1`] = `
<article
  className="overview"
>
  <h2
    className="lg-title"
  >
    Welcome to the logging app!
  </h2>
  <div>
    <p>
      You can create logs by pressing start or stop,
    </p>
    <p>
      view your logs by clicking on the View tab at the top,
    </p>
    <p>
      and highlight logs you want to adjust.
    </p>
  </div>
  <button
    className="lg-btn lg-btn-light mt-2"
    onClick={[Function]}
  >
     Next
  </button>
</article>
`;
